packages/core-data/src/hooks/test/use-entity-record.js

```diff
@@ -113,4 +113,38 @@ describe( 'useEntityRecord', () => {
 		expect( widget.editedRecord ).toEqual( { hello: 'foo', id: 1 } );
 		expect( widget.edits ).toEqual( { hello: 'foo' } );
 	} );
+
+	it( 'does not resolve entity record when disabled via options', async () => {
+		// Provide response
+		triggerFetch.mockImplementation( () => TEST_RECORD );
+
+		let data;
+		const TestComponent = () => {
+			data = useEntityRecord( 'root', 'widget', 2, {
+				options: { enabled: false },
+			} );
+			return <div />;
+		};
+		render(
+			<RegistryProvider value={ registry }>
+				<TestComponent />
+			</RegistryProvider>
+		);
+
+		expect( data ).toEqual( {
+			edit: expect.any( Function ),
+			editedRecord: {},
+			hasEdits: false,
+			edits: {},
+			record: null,
+			save: expect.any( Function ),
+		} );
+
+		// Fetch request should have been issued
+		await waitFor( () =>
+			expect( triggerFetch ).not.toHaveBeenCalledWith( {
```

Comment:
I think we should test what we expect and not something different. Probably use `expect( triggerFetch ).not.toHaveBeenCalled()`. The main thing though is that this test(even with my suggestion passes in trunk without your changes. 🤔 


By manually

Reply:
That's odd. This was failing without changes in my test. I'll have a loot at it tomorrow.

Reply:
I've updated the tests. They should fail on trunk but pass on this branch.

Reply:
This is really weird :). I also tried yesterday to reset the mocks etc.., and for me the issue remains, but I think it's something we miss with the testing library.

When I comment out your `(!enabled)` code and run the test, it still passes in trunk. When I'm changing though the second check to:
```
expect( triggerFetch ).toHaveBeenCalledWith( {
	path: '/wp/v2/widgets/2?context=edit',
} )
```
(removing the not). The tests works as expected, but the `expect( triggerFetch ).not.toHaveBeenCalled();` doesn't throw, which it should. I'll approve this PR now because the fix is good, but I'm curious why this is happening. I'll cc @kevin940726 and if we need to adjust something, let's do it in a follow up.

Reply:
I think `apiFetch` isn't dispatched when we do an assertion for the first time. But I wasn't able to figure out why. This seems to work for the first test.

cc @tyxla

Reply:
`waitFor` will immediately resolve if it passes the first time, so basically there's no waiting. This is tricky for testing something that "shouldn't" happen. Instead, I think we can fallback to the good old `setTimeout` and set an arbitrary reasonable waiting time.

```js
await act(
	() => new Promise( ( resolve ) => setTimeout( resolve, 0 ) )
);
expect( triggerFetch ).not.toHaveBeenCalled();
```

In this context, `setTimeout` of `0` works, but it really depends on the implementation details. Awaiting a microtask (promise) doesn't work here, for instance.

If we want to make this test more resilient to implementation details, then we can first make a "enabled" assertion (`.toHaveBeenCalled()`) with the `setTimeout(, 0)` approach then make another "disabled" test with the same `setTimeout(, 0)` setup.

Reply:
>waitFor will immediately resolve if it passes the first time, so basically there's no waiting.

It makes sense. Thank you!

Reply:
> If we want to make this test more resilient to implementation details, then we can first make a "enabled" assertion (.toHaveBeenCalled()) with the setTimeout(, 0) approach then make another "disabled" assertion with the same setTimeout(, 0) setup.

@kevin940726, can you elaborate bit more about this part? 🙇

Reply:
Sure! Because we want to make our unit tests as fast as possible, but we also don't want to just specify a random magic number (`0`) that could silently break when the implementation details change. Consider the below example:

```js
const TestComponent = ({ enabled }) => {
  data = useEntityRecord( 'root', 'widget', 2, { enabled } );
  return <div />;
};
const UI = ({ enabled }) => (
  <RegistryProvider value={ registry }>
    <TestComponent enabled={enabled} />
  </RegistryProvider>
);

// Smoke test.
const { rerender } = render(<UI enabled={true} />);
await act(() => new Promise(resolve => setTimeout(resolve, 0))); // The minimum delay.
expect(triggerFetch).toHaveBeenCalledTimes(1);

// Real test.
rerender(<UI enabled={false} />);
await act(() => new Promise(resolve => setTimeout(resolve, 0))); // The same delay.
expect(triggerFetch).toHaveBeenCalledTimes(1);
```

If the minimum delay (implementation detail) is still shorter than `setTimeout(,0)` then every thing works correctly and the second assertion is enough to test the feature as shown in the table below.

| Assertion passes? | First assertion | Second assertion |
|--------|--------|--------|
| `enabled:false` works | ✅ | ✅ |
| `enabled:false` breaks | ✅ | ❌ |

However, if the implementation detail changes to a longer delay (one animation frame for instance), the second assertion won't catch the regression and we'll need the first assertion (smoke test) to help us.

| Assertion passes? | First assertion | Second assertion |
|--------|--------|--------|
| `enabled:false` works | ❌ | ✅ |
| `enabled:false` breaks | ❌ | ✅ |

I think I'm overly complicating this now 😅. In short, the first assertion is just a smoke test to help us make sure the wait is enough for the hook to call `apiFetch` (when it should).